### PR TITLE
ast, parser: support create view syntax

### PR DESCRIPTION
**Pull request by XuHuaiyu**
Split #4874 into small PRs, 
this PR implements the support of MySQL CreateView syntax entirely.

PTAL @hanfei1991 @coocood

### CODE REVIEW

File: parser/parser.y [hunk 1]
```diff
@@ -2336,7 +2470,7 @@ UnReservedKeyword:
 | "SQL_NO_CACHE" | "DISABLE"  | "ENABLE" | "REVERSE" | "PRIVILEGES" | "NO" | "BINLOG" | "FUNCTION" | "VIEW" | "MODIFY" | "EVENTS" | "PARTITIONS"
 | "NONE" | "SUPER" | "EXCLUSIVE" | "STATS_PERSISTENT" | "ROW_COUNT" | "COALESCE" | "MONTH" | "PROCESS" | "PROFILES"
 | "MICROSECOND" | "MINUTE" | "PLUGINS" | "QUERY" | "SECOND" | "SHARE" | "SHARED" | "MAX_CONNECTIONS_PER_HOUR" | "MAX_QUERIES_PER_HOUR" | "MAX_UPDATES_PER_HOUR"
-| "MAX_USER_CONNECTIONS" | "REPLICATION" | "CLIENT" | "SLAVE" | "RELOAD" | "TEMPORARY" | "ROUTINE" | "EVENT"
+| "MAX_USER_CONNECTIONS" | "REPLICATION" | "CLIENT" | "SLAVE" | "RELOAD" | "TEMPORARY" | "ROUTINE" | "EVENT" | "ALGORITHM" | "DEFINER" | "INVOKER" | "MERGE" | "TEMPTABLE" | "UNDEFINED" | "SQL" | "SECURITY" | "CASCADED"
```
Comment (coocood) [inline parser/parser.y]: `SQL` is a reserved keyword.

File: parser/parser.y [hunk 2]
```diff
@@ -680,6 +692,12 @@ import (
 	ValuesOpt		"values optional"
 	VariableAssignment	"set variable value"
 	VariableAssignmentList	"set variable value list"
+	ViewAlgorithm		"view algorithm"
+	ViewCheckOption		"view check option"
+    ViewDefiner			"view definer"
```
Comment (coocood) [inline parser/parser.y]: alignment

File: parser/parser.y [hunk 3]
```diff
@@ -526,6 +536,7 @@ import (
 	ColumnDefList			"table column definition list"
 	ColumnName			"column name"
 	ColumnNameList			"column name list"
+	ColumnList              "column list"
```
Comment (coocood) [inline parser/parser.y]: alignment

**Comment by XuHuaiyu** [comment]
PTAL @coocood

**Comment by coocood** [comment]
LGTM

**Pull request merged**